agent/src/beerocks/slave/agent_db.h

```diff
@@ -72,8 +75,165 @@ class AgentDB {
     void db_lock() { m_db_mutex.lock(); }
     void db_unlock() { m_db_mutex.unlock(); }
 
+    /* Put down from here database members and functions used by the Agent modules */
+
 public:
-    /* Put here database members used by the Agent modules */
+    /* Agent Configuration */
+    struct sDeviceConf {
+        struct sFrontRadio {
+
+        } front_radio;
+
+        struct sBackRadio {
+
+        } back_radio;
+
+        bool local_gw;
+        bool local_controller;
+    } device_conf;
+
+    /** 
+     * Agent Sub Entities Data
+     */
+    struct sBridge {
+        sMacAddr mac;
+        std::string iface_name;
+    } bridge;
+
+    struct sBackhaul {
+        enum class eConnectionType { Invalid = 0, Wired, Wireless } connection_type;
+        std::string selected_iface_name;
+    } backhaul;
+
+    struct sEthernet {
+        sMacAddr mac;
+        std::string iface_name;
+    } ethernet;
+
+    struct sRadio {
+        sRadio(const std::string &front_iface_name, const std::string &back_iface_name)
+            : front(front_iface_name), back(back_iface_name)
+        {
+        }
+
+        struct sFront {
+            explicit sFront(const std::string &iface_name_)
+                : iface_name(iface_name_), max_supported_bw(eWiFiBandwidth::BANDWIDTH_UNKNOWN),
+                  freq_type(eFreqType::FREQ_UNKNOWN)
+            {
+            }
+            std::string iface_name;
+            sMacAddr iface_mac;
+            eWiFiBandwidth max_supported_bw;
+            eFreqType freq_type;
+
+            struct sBssid {
+                sMacAddr mac;
+                std::string ssid;
+                enum class eType { fAP, bAP } type;
+            };
+            std::array<sBssid, eBeeRocksIfaceIds::IFACE_TOTAL_VAPS> bssids;
+        } front;
+
+        struct sBack {
+            explicit sBack(const std::string &iface_name_) : iface_name(iface_name_) {}
+            std::string iface_name;
+            sMacAddr iface_mac;
+        } back;
+
+        struct sClient {
+            sClient(sMacAddr bssid_, size_t association_frame_length_, uint8_t *association_frame_)
+                : bssid(bssid_), association_time(std::chrono::steady_clock::now()),
```

Comment:
As I understood, this whole database structure is not testable (not mockable).
Specifically, performance-wise using the initializer list is better, especially when `sClient` could be constructed many time during the process lifetime.
Anyway, I'll be keeping it for now. If it will need to change in the future, we'll change it then.